### PR TITLE
XWIKI-15112 - Add an XWiki.AccessDenied customization option

### PR DESCRIPTION
**Pull request by slauriere**
JIRA reference: https://jira.xwiki.org/browse/XWIKI-15112

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentview.vm [hunk 1]
```diff
@@ -23,6 +23,9 @@ $services.progress.endStep()
 $services.progress.startStep('Display the title and content')
 <div class="xcontent">
   #template('contentheader.vm')
+  #foreach ($uix in $services.uix.getExtensions('org.xwiki.platform.template.content.header.after'))
+    $services.rendering.render($uix.execute(), 'xhtml/1.0')
+  #end
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentview.vm]: isn't that what you did already in the previous PR?

Comment (vmassol) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentview.vm]: what's the relationship with "XWIKI-15112 - Add an XWiki.AccessDenied customization option"?

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm [hunk 2]
```diff
@@ -12,15 +12,19 @@ $services.localization.render('notallowed')
...
 <div class="main layoutsubsection">
 <div id="mainContentArea">
-#if ($xwiki.isReadOnly())
-#xwikimessageboxstart($services.localization.render('error') $services.localization.render('readonly'))
+#if($xwiki.exists("XWiki.AccessDenied"))
+  $xwiki.includeForm("XWiki.AccessDenied",false)
 #else
-#xwikimessageboxstart($services.localization.render('error') $services.localization.render('notallowed'))
+  #if ($xwiki.isReadOnly())
+  #xwikimessageboxstart($services.localization.render('error') $services.localization.render('readonly'))
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm]: BTW, this should be indented two spaces (and same for the else part)

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm [hunk 3]
```diff
@@ -12,15 +12,19 @@ $services.localization.render('notallowed')
 #template("startpage.vm")
 <div class="main layoutsubsection">
 <div id="mainContentArea">
-#if ($xwiki.isReadOnly())
-#xwikimessageboxstart($services.localization.render('error') $services.localization.render('readonly'))
+#if($xwiki.exists("XWiki.AccessDenied"))
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm]: I think a space is missing after the `#if` and before the `(`. At least that's what you added below at line 18 ;)

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentview.vm (hunk 1 above)
Comment (slauriere) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentview.vm]: Indeed, I wrongly pushed a commit that related to another PR, sorry, I'll remove the change in the next commit.

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm [hunk 4]
```diff
@@ -6,21 +6,25 @@
...
+#if ("$!request.ajax" != '')
 $services.localization.render('notallowed')
 #else
 #template("startpage.vm")
 <div class="main layoutsubsection">
 <div id="mainContentArea">
-#if ($xwiki.isReadOnly())
-#xwikimessageboxstart($services.localization.render('error') $services.localization.render('readonly'))
+#if ($xwiki.exists("XWiki.AccessDenied"))
+  $xwiki.includeForm("XWiki.AccessDenied",false)
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm]: you're going to find that I nitpick (and you'll be right ;)), but could you add a space after `,` as in `, false`? ;)

**Comment by vmassol** [comment]
Sounds good. Discussing it on #xwiki to know if we can apply now.

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm (hunk 4 above)
Comment (slauriere) [reply xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/accessdenied.vm]: let me add a space indeed, I was about to do it in the first place and was hesitating

**Comment by vmassol** [comment]
thanks!

**Pull request merged**